### PR TITLE
docs: update shared example dependencies to Angular 7 and CLI 7

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Test run of example upgrades for V7

### CODE REVIEW

File: aio/content/examples/testing/src/index-specs.html [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
--->
+<!-- Add the app test styles-->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <base href="/">
+    <title>App Tests</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://unpkg.com/jasmine-core@2.99.1/lib/jasmine-core/jasmine.css">
```
Comment (IgorMinar) [inline aio/content/examples/testing/src/index-specs.html]: why not import the stylesheet from node_modules? cdn/external references like this are often source of CI flakes and also prevent the examples from working in the offline mode.

**Review by IgorMinar — APPROVED** [review]
please don't import the css from cdn. the rest looks good.

File: aio/tools/examples/README.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ Currently you will find the next boilerplates:
 * systemjs - Currently in deprecation, only used in a a few examples.
 * i18n - Based on the CLI one, features a few scripts for i18n.
 * universal - Based on the cli with a extra server for universal.
```
Comment (gkalpak) [inline aio/tools/examples/README.md]: While you're at it: _cli_ --> _CLI_ 🙏

File: aio/tools/examples/README.md [hunk 3]
```diff
@@ -32,6 +32,8 @@ Currently you will find the next boilerplates:
...
+* elements - Based on the CLI, with a custom `tsconfig.json` for changing the `target`
+* testing - Based on the CLI, with extra testing styles
```
Comment (gkalpak) [inline aio/tools/examples/README.md]: Is this still a thing?

**Review by gkalpak — COMMENTED** [review]
🎉 🎉 🎉 
(If you are into upgrading stuff, you might want to upgrade jasmine as well :wink:)

File: aio/content/examples/testing/src/index-specs.html (hunk 1 above)
Comment (gkalpak) [reply aio/content/examples/testing/src/index-specs.html]: These examples also need to work on StackBlitz (not sure `node_modules/` works there; they have `turbo_modules/`).

**Comment by gkalpak** [comment]
BTW, the testing examples ([sample app][1], [specs][2]) are broken on StackBlitz.
(The one is missing `environment.ts`, the other `polyfills.ts`; not sure why.)

[1]: https://pr25892-621d82a.ngbuilds.io/generated/live-examples/testing/stackblitz.html
[2]: https://pr25892-621d82a.ngbuilds.io/generated/live-examples/testing/specs.stackblitz.html

File: aio/content/examples/reactive-forms/e2e/src/app.e2e-spec.ts [hunk 4]
```diff
@@ -35,14 +35,14 @@ describe('Reactive forms', function () {
...
       await nameInput.sendKeys(nameText);
-      const value = await nameInput.getAttribute('value');
+      const value1 = await nameInput.getAttribute('value');
 
-      expect(value).toBe(nameText);
+      expect(value1).toBe(nameText);
       await updateButton.click();
 
-      const value = await nameInput.getAttribute('value');
+      const value2 = await nameInput.getAttribute('value');
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply aio/content/examples/reactive-forms/e2e/src/app.e2e-spec.ts]: Still looking into this one. Maybe due to TS upgrade. The error still was displayed in my IDE, but didn't fail the build

**Review by jenniferfell — APPROVED** [review]

File: aio/tools/examples/shared/boilerplate/cli/package.json [hunk 5]
```diff
@@ -28,16 +28,16 @@
...
+    "@angular/cli": "~7.0.0-rc.2",
     "@angular/compiler-cli": "~7.0.0-rc.0",
     "@angular/language-service": "~7.0.0-rc.0",
     "@types/jasmine": "~2.8.8",
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "~8.9.4",
     "codelyzer": "~4.3.0",
-    "jasmine-core": "~2.99.1",
-    "jasmine-marbles": "^0.3.1",
+    "jasmine-core": "^3.1.0",
```
Comment (gkalpak) [inline aio/tools/examples/shared/boilerplate/cli/package.json]: 👌 (OOC, why not 3.2.x?)

File: aio/tools/examples/shared/boilerplate/i18n/angular.json [hunk 6]
```diff
@@ -137,7 +137,7 @@
           "builder": "@angular-devkit/build-angular:protractor",
           "options": {
             "protractorConfig": "e2e/protractor.conf.js",
-            "devServerTarget": "angular.io-example:serve"
+            "devServerTarget": "angular.io-example:serve:fr"
```
Comment (gkalpak) [inline aio/tools/examples/shared/boilerplate/i18n/angular.json]: Why this change?

File: aio/content/examples/lazy-loading-ngmodules/src/app/app-routing.module.ts [hunk 7]
```diff
@@ -8,11 +8,11 @@ import { Routes, RouterModule } from '@angular/router';
 const routes: Routes = [
   {
     path: 'customers',
-    loadChildren: 'app/customers/customers.module#CustomersModule'
+    loadChildren: './customers/customers.module#CustomersModule'
```
Comment (gkalpak) [inline aio/content/examples/lazy-loading-ngmodules/src/app/app-routing.module.ts]: Is this a breaking change in Angular? Is it intentional? Documented?

File: aio/tools/examples/shared/boilerplate/universal/package.json [hunk 8]
```diff
@@ -5,13 +5,13 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "build": "ng build",
+    "build": "ng build --prod",
```
Comment (gkalpak) [inline aio/tools/examples/shared/boilerplate/universal/package.json]: Why this change? I thought the idea is to keep it as close as possible to what the cli would generate (unless there is a specific reason).

File: aio/tools/examples/shared/package.json [hunk 9]
```diff
@@ -82,7 +82,8 @@
     "ts-loader": "^4.2.0",
     "ts-node": "~7.0.0",
     "tslint": "~5.11.0",
-    "typescript": "~3.1.1"
+    "typescript": "~3.1.1",
+    "webpack-cli": "^2.0.14"
```
Comment (gkalpak) [inline aio/tools/examples/shared/package.json]: Why not use the latest version?

File: aio/content/examples/lazy-loading-ngmodules/src/app/app-routing.module.ts (hunk 7 above)
Comment (brandonroberts) [reply aio/content/examples/lazy-loading-ngmodules/src/app/app-routing.module.ts]: It is a breaking in how the `tsconfig.json` is generated with a new CLI project. I don't know if its documented as a breaking in the CLI. The router guide mentions the need for relative paths unless you modify `baseUrl` for your project.

File: aio/tools/examples/shared/boilerplate/i18n/angular.json (hunk 6 above)
Comment (brandonroberts) [reply aio/tools/examples/shared/boilerplate/i18n/angular.json]: This is the configuration for a V7 CLI project.

File: aio/tools/examples/shared/boilerplate/universal/package.json (hunk 8 above)
Comment (brandonroberts) [reply aio/tools/examples/shared/boilerplate/universal/package.json]: This was an error. Reverted

**Comment by brandonroberts** [comment]
@gkalpak should be ready for another review. I reverted the Jasmine version back to `~2.99.1` as that's what's generated with a new Angular CLI V7 project.

File: aio/tools/examples/shared/boilerplate/i18n/angular.json (hunk 6 above)
Comment (gkalpak) [reply aio/tools/examples/shared/boilerplate/i18n/angular.json]: I am not sure what you mean (what cli command generates a config file that uses `serve:fr` for e2e tests?), but since it was like this before I don't mind :grin:

**Review by gkalpak — APPROVED** [review]

**Review by gkalpak — DISMISSED** [review]
Actually, the live example seems to be broken.
